app/controllers/users.js

```diff
@@ -1,47 +1,139 @@
 import Ember from 'ember';
-import DS from 'ember-data';
 
-const {computed, inject, RSVP} = Ember;
-const {service} = inject;
-const {PromiseArray} = DS;
+const { computed, Controller, inject, isEmpty, PromiseProxyMixin, run } = Ember;
+const { service } = inject;
+const { debounce } = run;
+const { trim } = Ember.$;
+
+const ProxyContent = Ember.Object.extend(PromiseProxyMixin);
+
+function cleanQuery(query) {
+  return trim(query).replace(/[\-,?~!@#$%&*+\-'="]/, '');
+}
+
+export default Controller.extend({
+  init() {
+    this._super(...arguments);
+
+    // This needs to be removed after user session re-write
+    this.get('currentUser.model').then((user) => {
+      user.get('school').then((userSchool) => {
+        this.set('school', userSchool.get('id'));
+      });
+    });
+  },
+
+  queryParams: ['page'],
+  page: 1,
 
-export default Ember.Controller.extend({
   currentUser: service(),
   store: service(),
-  textFilter: '',
+
+  query: null,
+  delay: 500,
+
   limit: 20,
   offset: 0,
-  filteredUsers: computed('textFilter', function(){
-    let defer = RSVP.defer();
-    this.get('currentUser.model').then(user => {
-      user.get('school').then(school => {
-        if(this.get('textFilter')){
-          this.get('store').query('user', {
-            q: this.get('textFilter'),
-            school: school.get('id'),
-            limit: this.get('limit'),
-            offset: this.get('offset'),
-          }).then(users => {
-            defer.resolve(users);
-          });
-        } else {
-          this.get('store').query('user', {
-            school: school.get('id'),
-            limit: this.get('limit'),
-            offset: this.get('offset'),
-          }).then(users => {
-            defer.resolve(users);
-          });
-        }
+
+  prevPages: computed('page', {
+    get() {
+      const page = this.get('page');
+      return page !== 1;
+    }
+  }).readOnly(),
+
+  morePages: computed('model', {
+    get() {
+      const { model, limit } = this.getProperties('model', 'limit');
+      return model.length > limit;
+    }
+  }).readOnly(),
+
+  users: computed('model', {
```

Comment:
Instead of using the model here and having the complication of a workaround in the route why not just run the query here?  This data doesn't need to be resolved before the controller loads so it makes good sense to just do the work here in the controller instead of splitting half of it into the route.